src/authentication/middleware/index.ts

```diff
@@ -1,16 +1,31 @@
 import { NextFunction, Request, Response } from "express";
 import activeFeature from "../../feature.flag";
+import logger from "../../logger";
+import * as pageURLs from "../../model/page.urls";
 
 export default (req: Request, res: Response, next: NextFunction) => {
+  const referringPageURL = req.header("Referer") as string;
+
+  logger.debug("Check if user has referer");
+  if (referringPageURL === undefined) {
+    logger.debug("User has no referer - redirecting to index");
+    return res.redirect(pageURLs.EXTENSIONS);
+  }
+
+  logger.debug("Check if user is signed in");
   if (!req.chSession.isSignedIn()) {
-    let returnToUrl: string = "/extensions";
+
+    logger.debug("User not signed in");
+
+    let returnToUrl: string = pageURLs.EXTENSIONS;
     if (!activeFeature(process.env.ACCESSIBILITY_TEST_MODE)) {
-      const referringPageURL = req.header("Referer") as string;
-      if (req.originalUrl.endsWith("download")
-          || referringPageURL.endsWith("extensions")) {
+      // if user is coming from start page or download page
+      if (req.originalUrl.endsWith("/download")
+        || referringPageURL.endsWith(pageURLs.EXTENSIONS)) {
```

Comment:
So minor, but the double indentation for this line that was there before actually makes sense, because it's a continuation of the line and not a new 'block' of code.

Reply:
👍 will correct this